core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/jdbc/JDBCUserStoreManager.java

```diff
@@ -2784,52 +2787,62 @@ protected String preparePassword(String password, String saltValue) throws UserS
     }
 
     /**
-     * Prepare the password including the salt, and hashes if hash algorithm is provided
+     * Prepare the password including the salt, and hashes if hash algorithm is provided.
      *
-     * @param password original password value
-     * @param saltValue salt value
-     * @return  hashed password or plain text password as a String
-     * @throws UserStoreException
+     * @param password  Original password value which needs to be hashed.
+     * @param saltValue Salt value.
+     * @return Hashed password or plain text password as a String.
+     * @throws UserStoreException The exception thrown at hashing the passwords.
      */
     protected String preparePassword(Object password, String saltValue) throws UserStoreException {
 
         Secret credentialObj;
+        String passwordHash = StringUtils.EMPTY;
         try {
             credentialObj = Secret.getSecret(password);
         } catch (UnsupportedSecretTypeException e) {
             throw new UserStoreException("Unsupported credential type", e);
         }
-
-        try {
-            String passwordString;
-            if (saltValue != null) {
-                credentialObj.addChars(saltValue.toCharArray());
-            }
-
-            String digestFunction = realmConfig.getUserStoreProperties().get(JDBCRealmConstants.DIGEST_FUNCTION);
-            if (digestFunction != null) {
-                if (digestFunction.equals(UserCoreConstants.RealmConfig.PASSWORD_HASH_METHOD_PLAIN_TEXT)) {
-                    passwordString = new String(credentialObj.getChars());
-                    return passwordString;
+        Map<String, Object> userStoreProperties = Collections.unmodifiableMap(realmConfig.getUserStoreProperties());
```

Comment:
Why do we need `unmodifiableMap` map? Shall we use that only to pass the `userStoreProperties` to the service?